rts/motoko-rts/src/gc/mark_compact.rs

```diff
@@ -108,11 +102,12 @@ unsafe fn mark_static_roots<M: Memory>(mem: &mut M, static_roots: SkewedPtr, hea
     // Static objects are not in the dynamic heap so don't need marking.
     for i in 0..root_array.len() {
         let obj = root_array.get(i).unskew() as *mut Obj;
-        mark_fields(mem, obj, heap_base);
+        mark_fields(mem, obj, obj.tag(), heap_base);
```

Comment:
An optimization I've done in #2647: we know `obj` here is a `MutBox` so we can check that with a `debug_assert` and inline and specialize `mark_fields` for `MutBox`.

Reply:
I implemented this.